### PR TITLE
KPED + GIT galaxies

### PR DESCRIPTION
**Pull request by mcoughlin**
Auto-increment the tiles when doing galaxy targeting

### CODE REVIEW

**Comment by lpsinger** [comment]
What about an auto increment column in the database?

**Comment by mcoughlin** [comment]
@lpsinger it also needs to match what is put into PlannedObservation (as you see at the bottom). so the mapping from the original to the new one needs to be done (this is trivial as just an addition for this one).

**Comment by lpsinger** [comment]
Yes, and why not accomplish that with an auto-incrementing column?

**Comment by mcoughlin** [comment]
Also won't this trigger a rebuild of the database (to set autoincrement=True)?!? I am ok with changing this, but it seems easiest how I have done it.

**Comment by lpsinger** [comment]
> Also won't this trigger a rebuild of the database (to set autoincrement=True)?!?

Yes, but a straightforward one.

> I am ok with changing this, but it seems easiest how I have done it.

It's easier this way, yes, but it will crash if the marshal tries to save two observing plans at once.

**Comment by mcoughlin** [comment]
field = models.Field(telescope=telescope,
                                         ra=ra, dec=dec,
                                         contour=contour,
                                         reference_filter_ids=ref_filter_ids,
                                         reference_filter_mags=ref_filter_mags,
                                         ipix=ipix.tolist())
                    models.db.session.merge(field)

so to access this field now though, you want the field_id yes? but you don't know what it is? do we need to cross-match with ra and dec?

**Comment by lpsinger** [comment]
Could we back up a minute here, and would you please explain to me what is currently wrong with KPED or GIT galaxy tiling?

File: growth/too/tasks/tiles.py [hunk 1]
```diff
@@ -264,7 +271,18 @@ def get_planned_observations(
         field_ids, ras, decs, probs, nexposures = [], [], [], [], []
         segmentlist = segments.segmentlist()
         totprob = 0.0
-        for field_id in tile_structs[telescope].keys():
+
+        field_maps = {}
+
+        if params["tilesType"] == "galaxy":
+            fields = models.Field.query.filter_by(telescope=telescope).all()
+            field_id_max = -1
```
Comment (lpsinger) [inline growth/too/tasks/tiles.py]: Is field_id_max used? It doesn't seem to be.

File: growth/too/models.py [hunk 2]
```diff
@@ -435,8 +437,11 @@ class Field(db.Model):
 
     field_id = db.Column(
         db.Integer,
+        Sequence('field_id_seq', start=10000001, increment=1),
```
Comment (lpsinger) [inline growth/too/models.py]: Is it possible to count down from zero to avoid any possible collision with pre-existing fields?

**Comment by mcoughlin** [comment]
@lpsinger is this still good to merge? I totally lost track of this one, sorry.

**Comment by lpsinger** [comment]
Please rebase it. We need to check that the unit tests pass.

**Comment by mcoughlin** [comment]
@lpsinger Rebased and unit tests are passing. Recall that the Fields database has the Sequence added to it now, so I am unsure what exactly needs doing when it is deployed.

**Comment by lpsinger** [comment]
This pull request now does a lot more than just fixing the database auto-increment issue. Would you please break it into logically separate pull requests?

**Pull request closed without merging**